remove-apache-license-identifier.py

```diff
@@ -0,0 +1,184 @@
+    license. All files in the starter kit contained references to the
+    Apache license. The starter kit installation scripts copied files
+    from the stater kit into the project repository.  This became an
+    issue when the project repository was released under a different
+    license.  This script removes all references to the Apache license
+    from the files copied into the project repository from the starter
+    kit.
+    """
+
+    return create_parser(desc, args, epilog)
+
+def configure_logging(args):
+    # Logging is configured by the first invocation of logging.basicConfig
+    fmt = '%(levelname)s: %(message)s'
+    if args.debug:
+        logging.basicConfig(level=logging.DEBUG, format=fmt)
+    if args.verbose:
+        logging.basicConfig(level=logging.INFO, format=fmt)
+    logging.basicConfig(format=fmt)
+
+################################################################
+# Shell out commands
+
+def run(cmd, cwd=None, encoding=None):
+    """Run a command in a subshell and return the standard output.
+
+    Run the command cmd in the directory cwd and use encoding to
+    decode the standard output.
+    """
+
+    kwds = {
+        'cwd': cwd,
+        'stdout': subprocess.PIPE,
+        'stderr': subprocess.PIPE,
+        'text': True,
+    }
+    if sys.version_info >= (3, 6): # encoding introduced in Python 3.6
+        kwds['encoding'] = encoding
+
+    logging.debug('Running "%s" in %s', ' '.join(cmd), cwd)
+
+    result = subprocess.run(cmd, **kwds, check=False)
+    if result.returncode:
+        logging.debug('Failed command: %s', ' '.join(cmd))
+        logging.debug('Failed return code: %s', result.returncode)
+        logging.debug('Failed stdout: %s', result.stdout.strip())
+        logging.debug('Failed stderr: %s', result.stderr.strip())
+        return []
+
+    # Remove line continuations before splitting stdout into lines
+    # Running command with text=True converts line endings to \n in stdout
+    lines = result.stdout.replace('\\\n', ' ').splitlines()
+    return [strip_whitespace(line) for line in lines]
+
+def strip_whitespace(string):
+    return re.sub(r'\s+', ' ', string).strip()
+
+################################################################
+
+def maybe_a_copied_file(path):
+    copied_files = [
+        'Makefile',
+        'Makefile-project-defines',
+        'Makefile-project-targets',
+        'Makefile-project-testing',
+    ]
+    return os.path.basename(path) in copied_files
+
+def find_apache_references(proofdir=None):
+    paths = run(['git', 'grep', '-l', 'Apache', '.'], cwd=proofdir)
+    paths = [os.path.normpath(os.path.join(proofdir, path)) for path in paths]
+    paths = [path for path in paths if not os.path.islink(path)]
+    return sorted(paths)
+
+def remove_apache_reference(path, extension='backup'):
+    apache_references = [
+        '# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.',
```

Comment:
Why are we removing this line? I would assume that we only want to get rid of the SPDX line below, not the copyright declaration.

Also, why are we deleting lines? Isn't the point of this script to change Apache into MIT (which is what the rest of this PR actually does)?

I wonder whether it might be more convenient for this script to just replace `Apache-2.0` in the relevant files with an SPDX string of the user's choice, defaulting to `MIT-0`?

Reply:
This is the proposal worked out in discussion with AWS IoT and AWS Open Source.